### ISSUE TITLE
Example still doesn't render map

### ISSUE THREAD

**Issue by alubling** [nitaliano/react-native-mapbox-gl#160]
So i also tried the example and got a white screen, although the text renders. 

<img width="373" alt="screen shot 2015-11-29 at 3 19 21 pm" src="https://cloud.githubusercontent.com/assets/7263959/11459790/98adc448-96ac-11e5-8781-84bbf9db5179.png">

Read through open and closed issues because I was getting similar errors, but none of those fixes worked. 
1. Installed carefully per the docs
2. Copy and pasted the example code
3. Put in my token
4. Changed {this.mapStyles.emerald} to {this.mapStyles.streets} (also tried my personal map)

Get the commonly mentioned error: Failed propType: typeChecker is not a function Check the render method of 'MapExample'

Reducing to just:

``` javascript
<Mapbox
      style={styles.map}
      accessToken={'my_token'}
      styleURL={this.mapStyles.streets} />
```

gets rid of the error, but the result is the same, map doesn't render. 

Tried to fix this for way too long, could use any help. Thanks, excited to get this working!

**Comment by rt2zz** [comment]
I am seeing the same thing (i.e. nothing)

**Comment by mihaigiurgeanu** [comment]
I have the same error. Nothing is displayed in IOS version. The Android version works good.

**Comment by bsudekum** [comment]
Anything in the console? Did you add `mixins`? This is usually do to a bad installation, I just ran through it and the map appeared just fine.

**Comment by alubling** [comment]
The console is clean. The only mixin in the code is Mapbox.mixin. I'll try another re-install. Is there any versioning issue maybe? 

Thanks for the quick response btw.

**Comment by bsudekum** [comment]
> Is there any versioning issue maybe?

Double check that your installing v2.0.0

**Comment by alubling** [comment]
Yep, that's the one. If i do another install do i need to wipe out anything from xcode explicitly?

**Comment by mihaigiurgeanu** [comment]
It is v2.0.0, but, when installing, I get an error for 2 components,  bufferutil@1.2.1 and utf-8-validate@1.2.1, about the missing libgcc_s.10.5. I have Xcode 7 from the beginning (it is not an update from Xcode 6, I directly  installed Xcode 7) and none of the fixes I found on the net worked for me. I am thinking, maybe, because of this error, the app would not work correctly.

This is the error:

```
> bufferutil@1.2.1 install /Users/mihaigiurgeanu/Projects/Example/node_modules/bufferutil
> node-gyp rebuild

  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
  SOLINK_MODULE(target) Release/bufferutil.node
ld: library not found for -lgcc_s.10.5
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Release/bufferutil.node] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/mihaigiurgeanu/.nvm/versions/node/v5.1.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.0.0
gyp ERR! command "/Users/mihaigiurgeanu/.nvm/versions/node/v5.1.0/bin/node" "/Users/mihaigiurgeanu/.nvm/versions/node/v5.1.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/mihaigiurgeanu/Projects/Example/node_modules/bufferutil
gyp ERR! node -v v5.1.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
npm WARN install:bufferutil@1.2.1 bufferutil@1.2.1 install: `node-gyp rebuild`
npm WARN install:bufferutil@1.2.1 Exit status 1

> utf-8-validate@1.2.1 install /Users/mihaigiurgeanu/Projects/Example/node_modules/utf-8-validate
> node-gyp rebuild

  CXX(target) Release/obj.target/validation/src/validation.o
  SOLINK_MODULE(target) Release/validation.node
ld: library not found for -lgcc_s.10.5
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Release/validation.node] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/mihaigiurgeanu/.nvm/versions/node/v5.1.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.0.0
gyp ERR! command "/Users/mihaigiurgeanu/.nvm/versions/node/v5.1.0/bin/node" "/Users/mihaigiurgeanu/.nvm/versions/node/v5.1.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/mihaigiurgeanu/Projects/Example/node_modules/utf-8-validate
gyp ERR! node -v v5.1.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
npm WARN install:utf-8-validate@1.2.1 utf-8-validate@1.2.1 install: `node-gyp rebuild`
npm WARN install:utf-8-validate@1.2.1 Exit status 1
```

**Comment by alubling** [comment]
@bsudekum - k did another full re-install process, still getting the same result.

**Comment by dagatsoin** [comment]
Just made mine worked on iphone4s simulator.
I have node 4.2.2 and here are my dependencies: 

```
    "react-native": "^0.15.0",
    "react-native-mapbox-gl": "^2.0.0",
```

**Comment by alubling** [comment]
@dagatsoin - tried the same dependencies, still not working for me.

Although the screen is white i can still get a console.error "Argument 0 (NSNumber) of RCTMapboxGLManager.setVisibleCoordinateBoundsAnimated must not be null" by panning on the screen.

**Comment by bsudekum** [comment]
@alubling what version of react native are you running?

**Comment by alubling** [comment]
@bsudekum - 0.15.0

**Comment by dagatsoin** [comment]
did you tried with different version of node.js/npm (for me node 4.2.2 and npm 2)

**Comment by alubling** [comment]
@dagatsoin - tried with 4.2.1, then after your comment i upgraded to 4.2.2, which also doesn't work.

**Comment by dagatsoin** [comment]
I think you should share you entire repo with the exact version of all you global tools (node, Xcode, etc...) because there is nothing to hook on for now.

**Comment by alubling** [comment]
@bsudekum @dagatsoin - thanks for the help guys. Got it working. Going to close this but just in case someone opens this thread:
1. Removed everything in the render function other than the map, and stripped that down to just the required props. 
2. Swapped out the existing style options for a custom one that i created using mapbox studio, otherwise the screen was black, regardless of the option. 
3. I also moved from the simulator to the phone, not sure if that had anything to do with it though.
